src/app.js

```diff
@@ -58,33 +83,104 @@ export function app(props, container) {
 
     assign(state, module.state)
 
-    initActions(state, actions, module.actions)
-
-    for (var i in module.modules) {
-      init(module.modules[i], (state[i] = {}), (actions[i] = {}))
+    initModuleActions(state, actions, module.actions, update, getState)
+
+    for (var key in module.modules) {
+      initModule(
+        module.modules[key],
+        // do not override state is already exist in current module
+        state[key] || (state[key] = {}),
+        // do not override actions is already exist in current module
+        actions[key] || (actions[key] = {}),
```

Comment:
Should solve #438.

Reply:
@Mytrill My PR doesn't have this, is it necessary?